ingest/anndata_.py

```diff
@@ -155,7 +155,14 @@ def generate_processed_matrix(adata):
         h5ad_frag.features.processed.tsv
         Gzip files for faster delocalization
         """
-        pd.DataFrame(adata.var.index).to_csv(
+        if adata.var.index.name == 'gene_ids':
+            # CELLxGENE indexes by Ensembl gene ID, not gene name (i.e. symbol).
+            # Gene name is encoded in feature_name, which is needed for gene search.
+            feature_frame = adata.var.feature_name
```

Comment:
Ingest's [mtx processing will extract the Ensembl ID](https://github.com/broadinstitute/scp-ingest-pipeline/blob/79356b955276d016ec8d75d17b57071207a53045/ingest/expression_files/mtx.py#L265) and store it with the gene name (so the gene expression data CAN be searchable by both ensembl ID AND gene name) if we generate a two column features file. 

Leave the original "pd.DataFrame(adata.var.index).to_csv" for the else clause and give something like this a try in the if clause:

df[['var_names', 'feature_name']].to_csv(
            "h5ad_frag.features.processed.tsv.gz",
            sep="\t",
            index=False,
            header=False,
            compression="gzip",
        )

Reply:
That approach didn't work on my end, but (after some trial-and-error) [preserving the index](https://github.com/broadinstitute/scp-ingest-pipeline/pull/341/commits/17f23e9abb3410386f8a736d2b0b12098920adaf) did.

Per chat, search by gene ID hasn't worked in a long time, so this won't add user value right now.  But, when that's restored, this handling might be needed for alternative search if a study uses a bespoke gene annotation.

I've updated the automated test, and confirmed by manually uploading again to the study on staging.